File: project_java/src/main/java/hu/bme/aut/onlab/dxhjoh/couchsurfing_java_backend/repository/UserRepository.java
package hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.repository;

import hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.model.User;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {
    Optional<User> findByUsername(String username);

    boolean existsByUsernameIgnoreCase(String username);

    boolean existsByEmail(String email);
}

File: project/src/main/kotlin/hu/bme/aut/onlab/couchsurfing_backend/repository/UserRepository.kt
package hu.bme.aut.onlab.couchsurfing_backend.repository

import hu.bme.aut.onlab.couchsurfing_backend.model.User
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository: JpaRepository<User, Int> {
    fun findByEmail(email: String): User

    fun findByUsername(username: String): User
}